src/storage.rs

```diff
@@ -37,19 +37,98 @@ async fn collect_body(
 pub enum InvalidNameError {
     /// The given name did not have a valid bucket
     #[error("error validating bucket name {1}")]
-    Bucket(#[source] api::Error, String),
+    Bucket(#[source] BucketNameError, String),
```

Comment:
To better support adopting tame-gcs's fix (hoping one happens), can we keep the error type the same? tame_gcs::Error has public variants, it should be possible to construct the same `Invalid*` variants

Reply:
Good point, I've switched back to `tame-gcs` errors.